core/arch/x86/encode.c

```diff
@@ -1819,6 +1819,13 @@ encode_operand(decode_info_t *di, int optype, opnd_size_t opsize, opnd_t opnd)
              * This only works if the instr has no other immeds!
              */
             instr_t *target_instr = opnd_get_instr(opnd);
+            /* XXX: This check could be inaccurate, say if the first instruction in
+             * the instrlist containted the noted near instr operand.
```

Comment:
Yeah I'm thinking we should just remove this, if we have the DOCHECK.  It's not easy to make it have no false positives.

("containted", "note == NULL")